include/rocksdb/utilities/table_properties_collectors.h

```diff
@@ -5,12 +5,59 @@
 
 #pragma once
 #ifndef ROCKSDB_LITE
+#include <atomic>
 #include <memory>
 
 #include "rocksdb/table_properties.h"
 
 namespace rocksdb {
 
+// A factory of a table property collector that marks a SST
+// file as need-compaction when it observe at least "D" deletion
+// entries in any "N" consecutive entires.
+class CompactOnDeletionCollectorFactory
+    : public TablePropertiesCollectorFactory {
+ public:
+  virtual ~CompactOnDeletionCollectorFactory() {}
+
+  virtual TablePropertiesCollector* CreateTablePropertiesCollector(
+      TablePropertiesCollectorFactory::Context context) override;
+
+  // Change the value of sliding_window_size "N"
```

Comment:
is there a way to use this as a kill-switch? e.g., by setting sliding_window_size to zero or negative? I recall that was something Mikhail mentioned to me.

Reply:
Oh, it didn't come up in our conversation, but that sounds like a good idea. There is currently no way to turn it off after opening the DB. Setting sliding_window_size to 0 can be the kill switch.